seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxDBSourceSplitEnumerator.java

```diff
@@ -0,0 +1,161 @@
+        this.pendingSplit = new HashSet<>();
+    }
+
+    @Override
+    public void run() throws Exception {
+        pendingSplit = getInfluxDBSplit();
+        assignSplit(context.registeredReaders());
+    }
+
+    @Override
+    public void close() throws IOException {
+        //nothing to do
+    }
+
+    @Override
+    public void addSplitsBack(List splits, int subtaskId) {
+        if (!splits.isEmpty()) {
+            pendingSplit.addAll(splits);
+            assignSplit(Collections.singletonList(subtaskId));
+        }
+    }
+
+    @Override
+    public int currentUnassignedSplitSize() {
+        return pendingSplit.size();
+    }
+
+    @Override
+    public void handleSplitRequest(int subtaskId) {
+        //nothing to do
+    }
+
+    @Override
+    public void registerReader(int subtaskId) {
+        if (!pendingSplit.isEmpty()) {
+            assignSplit(Collections.singletonList(subtaskId));
+        }
+    }
+
+    @Override
+    public InfluxDBSourceState snapshotState(long checkpointId) throws Exception {
+        return new InfluxDBSourceState(assignedSplit);
+    }
+
+    @Override
+    public void notifyCheckpointComplete(long checkpointId) throws Exception {
+        //nothing to do
+
+    }
+
+    private Set<InfluxDBSourceSplit> getInfluxDBSplit() {
+        String sql = config.getSql();
+        Set<InfluxDBSourceSplit> influxDBSourceSplits = new HashSet<>();
+        // no need numPartitions, use one partition
+        if (config.getPartitionNum() == 0) {
+            influxDBSourceSplits.add(new InfluxDBSourceSplit(InfluxDBConfig.DEFAULT_PARTITIONS, sql));
+            return influxDBSourceSplits;
+        }
+        long start = config.getLowerBound();
+        long end = config.getUpperBound();
+        int numPartitions = config.getPartitionNum();
+        String[] sqls = sql.split(SQL_WHERE);
+        if (sqls.length > 2) {
+            throw new IllegalArgumentException("sql should not contain more than one where");
+        }
+        int size = (int) (end - start) / numPartitions + 1;
+        int remainder = (int) ((end + 1 - start) % numPartitions);
+        if (end - start < numPartitions) {
+            numPartitions = (int) (end - start);
+        }
+        long currentStart = start;
+        int i = 0;
+        while (i < numPartitions) {
+            String query = " where (" + config.getSplitKey() + " >= " + currentStart + " and " + config.getSplitKey()  + " < " + (currentStart + size) + ") ";
+            i++;
+            currentStart += size;
+            if (i + 1 <= numPartitions) {
+                currentStart = currentStart - remainder;
+            }
+            query = sqls[0] + query;
+            if (sqls.length > 1) {
+                query = query + " and ( " + sqls[1] + " ) ";
+            }
+            influxDBSourceSplits.add(new InfluxDBSourceSplit(String.valueOf(i + System.nanoTime()), query));
+        }
+        return influxDBSourceSplits;
+    }
```

Comment:
extract this piece of logic as a public method? WDYT?

Reply:
> extract this piece of logic as a public method? WDYT?

It's a good suggestion. to abstract a public method to seatunnel-common module?